### ISSUE TITLE
IgxList: missing export after update from v15 to v16

### ISSUE THREAD

**Issue by dmk1111** [IgniteUI/igniteui-angular#13484]
## Description  
We use IGX components in our project but have our own wrappers for consistency according to design guidelines. After updating from Angular 15 to 16 and IGX components to latest version as well, we found out that `IgxListItem` component doesn't work correctly because we can't provide `IgxListBaseDirective` anymore - it was removed from exports and this breaking change was not noted anywhere

 * @infragistics/igniteui-angular version: 16.0.23
 *  previous @infragistics/igniteui-angular version: 15.0.7

## Steps to reproduce  

1. Create wrappers around `igx-list` - ListComponent and `igx-list-item` - ListItemComponent;
2. inside ListItemComponent add 

```ts
providers: [
    {
      provide: IgxListBaseDirective,
      useExisting: ListComponent,
    },
  ],
```

## Result  
Can not find `IgxListBaseDirective` in `@infragistics/igniteui-angular`

## Expected result  
We can use components in our code without errors:
```html
<app-list>
  <app-list-item></app-list-item>
</app-list>
```

**Comment by kdinev** [comment]
@dmk1111 Thank you for your contribution! I suppose you would want the export in a patch in `16.0.x`?

**Comment by dmk1111** [comment]
@kdinev yes, please, if possible. We're using **patch-package** as temporary solution for now

**Comment by kdinev** [comment]
@dmk1111 Could you cherry-pick your commit and open a second PR against `16.0.x`. That would be the one that would go out with the patch, because master is tracking `16.1.x` at the moment :)

**Comment by dmk1111** [comment]
@kdinev thanks, created separate PR: #13489

**Comment by kdinev** [comment]
@dmk1111 The change will be released in `16.0.25` :)
